starcompactor/extractors/mysql.py

```diff
@@ -38,24 +39,21 @@ def traces_query(db, start=None, end=None):
         i.project_id,
         i.hostname,
         i.host,
-        -- ia.created_at,
-        -- ia.id AS action_id,
-        -- iae.id AS event_id,
         iae.event,
         iae.result,
         iae.start_time,
         iae.finish_time
     FROM
-        nova.instances AS i
+        {database_name}.instances AS i
             JOIN
-        nova.instance_actions AS ia ON i.uuid = ia.instance_uuid
+        {database_name}.instance_actions AS ia ON i.uuid = ia.instance_uuid
             JOIN
-        nova.instance_actions_events AS iae ON ia.id = iae.action_id
+        {database_name}.instance_actions_events AS iae ON ia.id = iae.action_id
     WHERE
         i.user_id != 'admin'
         AND
         i.project_id != 'admin'
-    '''
+    '''.format(database_name=database_name)
```

Comment:
This is vulnerable to injection, but I don't know of a great way to make it not that way. Is there a way to run `query()` where you pass it keyword args for it to template in itself (presumably in a safe way)? Then you could do like `FROM %(database_name).instances ...`

Reply:
seems like database name or table name can not be passed as parameter.